src/utils.ts

```diff
@@ -62,25 +65,77 @@ export const getW3UpClient = async ({
 };
 
 interface GetS3ClientParams {
-    endpoint: string;
+    s3Bucket: string;
     accessKeyId: string;
     secretAccessKey: string;
 }
 
-export const getS3Client = ({
-    endpoint,
+export const getS3Client = async ({
+    s3Bucket,
     accessKeyId,
     secretAccessKey,
-}: GetS3ClientParams): S3Client => {
-    return new S3Client({
-        forcePathStyle: false, // Configures to use subdomain/virtual calling format.
-        endpoint,
+}: GetS3ClientParams): Promise<S3Client> => {
+    const client = new S3Client({
         region: "us-east-1",
         credentials: {
             accessKeyId,
             secretAccessKey,
         },
     });
+
+    const putBucketLifecycleConfiguration =
```

Comment:
It's just to make sure it's always there. Since this full lifecycle and its whole concept is managed by this specific service it makes sense that it's created here instead of Terraform no?